test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -2068,24 +2069,29 @@ var _ = ginkgo.Describe("Interacting with scheduler", ginkgo.Ordered, ginkgo.Con
 
 var _ = ginkgo.Describe("Job controller interacting with Workload controller when waitForPodsReady with requeuing strategy is enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
 	var (
-		backoffBaseSeconds int32
-		backoffLimitCount  *int32
-		ns                 *corev1.Namespace
-		fl                 *kueue.ResourceFlavor
-		cq                 *kueue.ClusterQueue
-		lq                 *kueue.LocalQueue
+		backoffBaseSeconds      int32
+		backoffLimitCount       *int32
+		waitForPodsReadyTimeout *metav1.Duration
+		ns                      *corev1.Namespace
+		fl                      *kueue.ResourceFlavor
+		cq                      *kueue.ClusterQueue
+		lq                      *kueue.LocalQueue
+		wl                      kueue.Workload
+		wlKey                   types.NamespacedName
+		job                     *batchv1.Job
 	)
 
 	ginkgo.JustBeforeEach(func() {
 		waitForPodsReady := &configapi.WaitForPodsReady{
 			Enable:         true,
 			BlockAdmission: ptr.To(true),
-			Timeout:        &metav1.Duration{Duration: util.TinyTimeout},
+			Timeout:        waitForPodsReadyTimeout,
 			RequeuingStrategy: &configapi.RequeuingStrategy{
 				Timestamp:          ptr.To(configapi.EvictionTimestamp),
 				BackoffBaseSeconds: ptr.To(backoffBaseSeconds),
 				BackoffLimitCount:  backoffLimitCount,
 			},
+			RecoveryTimeout: &metav1.Duration{Duration: util.TinyTimeout},
```

Comment:
I would suggest a separate config (probably Describe?) for that. Yes, this is more code, but:
- testing without `RecoveryTimeout` makes sense as this remains a valid configuration
- it is hard for reviewers to decouple which changes in the files are due to logic changes and which due to the introduction of the timeout.

Reply:
cc @tenzen-y